crates/re_log_types/src/data_cell.rs

```diff
@@ -295,6 +299,8 @@ impl DataCell {
     /// See [`Self::try_from_arrow`] for the fallible alternative.
     #[inline]
     pub fn from_arrow(name: ComponentName, values: Box<dyn arrow2::array::Array>) -> Self {
+        // NOTE: see function description why it's okay here
+        #[allow(clippy::unwrap_used)]
```

Comment:
Deserialization could fail for anything, including built-in components.

We should remove the naked `from_arrow` method instead.

Reply:
I'm not sure I understand correctly - you mean completely remove `DataCell::from_arrow`, `from_arrow_empty`, `to_native`, etc. and then refactor their usage everywhere so their `try_...` versions are used instead?

Reply:
yes, ideally. It may be a lot of work though, which is likely why no one has done it yet

Reply:
Haha, makes sense. I can do it but maybe leave it as it is in this PR and make a separate one - if that's okay for you?

Reply:
definetly!

Reply:
Okay, I will create a separate ticket for the `from_*` functions.
I also rebased so it should be mergable if there is nothing else?